compiler/test/compilable/named_arguments.d

```diff
@@ -0,0 +1,61 @@
+
+string fun(string x, string y, string z = "Z", string w = "W")
+{
+	return x ~ y ~ z ~ w;
+}
+
+static assert(fun(   "x",    "y") == "xyZW");
+static assert(fun(   "x",    "y", "z", "w") == "xyzw");
+static assert(fun(x: "x", y: "y", z: "z", w: "w") == "xyzw");
+static assert(fun(w: "w", z: "z", y: "y", x: "x") == "xyzw");
+static assert(fun(y: "y",    "z", x: "x") == "xyzW");
+static assert(fun(   "x",    "y", w: "w") == "xyZw");
```

Comment:
static assert(fun( x:"x", "y", z:"z") == "xyzW");

Reply:
Also, with named arguments, default parameters needn't be only at the end. You could have:
```
int fun(x, y = 5, z)
```
and call it with:
```
fun(x:1, z:2);
```

Reply:
Yes, that's follow up work (along with named template arguments and IFTI support)

Reply:
> static assert(fun( x:"x", "y", z:"z") == "xyzW");

I added the test case